### PR TITLE
Enable knative serving test

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- enable knative serving test

**Related issue(s)**

#3498

### CODE REVIEW

File: resources/knative-serving/values.yaml [hunk 1]
```diff
@@ -10,8 +10,8 @@ global:
   containerRegistry:
     path: eu.gcr.io/kyma-project
   test_knative_serving_acceptance:
-    dir: develop/
-    version: 4a2b51ec
+    dir: pr/
+    version: PR-3657
```
Comment (sayanh) [inline resources/knative-serving/values.yaml]: An image with a PR tag should not be merged to master.

Comment (k15r) [reply resources/knative-serving/values.yaml]: i know, i will update the PR as soon as i have that. I wanted to create a draft PR, but forgot during creation and GH does not allow you to do this afterwards.

but blocking the merge is a good idea, so i cannot forget it. THX 😄

File: resources/knative-serving/templates/tests/test-serving.yaml [hunk 2]
```diff
@@ -1,22 +1,83 @@
...
+---
+{{- if .Capabilities.APIVersions.Has "testing.kyma-project.io/v1alpha1" }}
+apiVersion: "testing.kyma-project.io/v1alpha1"
+kind: TestDefinition
+metadata:
+  name: test-knative-serving-acceptance
+template:
+  metadata:
+    annotations:
+      sidecar.istio.io/inject: "true"
```
Comment (aszecowka) [inline resources/knative-serving/templates/tests/test-serving.yaml]: sidecar injection rules should be the same in both cases

File: resources/knative-serving/templates/tests/test-serving.yaml [hunk 3]
```diff
@@ -1,22 +1,83 @@
...
+  spec:
```
Comment (aszecowka) [inline resources/knative-serving/templates/tests/test-serving.yaml]: Containers shoulf be defined under: spec.template.spec, here you have an example:
```
apiVersion: testing.kyma-project.io/v1alpha1
kind: TestDefinition
metadata:
  name: test-sth
spec:
  template:
    spec:
      shareProcessNamespace: true
      restartPolicy: Never
      containers:
        - name: test
          image: alpine:latest
```

File: resources/knative-serving/templates/tests/test-serving.yaml (hunk 2 above)
Comment (k15r) [reply resources/knative-serving/templates/tests/test-serving.yaml]: good catch

File: resources/knative-serving/templates/tests/test-serving.yaml (hunk 3 above)
Comment (k15r) [reply resources/knative-serving/templates/tests/test-serving.yaml]: ah, yes that make more sense

**Pull request merged**